File: src/main/java/io/openfuture/openmessanger/repository/entity/ChatParticipant.java
package io.openfuture.openmessanger.repository.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor

@Entity
@Table(name = "chat_participant")
public class ChatParticipant {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private Integer chatId;

    private String username;

    public ChatParticipant(final Integer chatId, final String username) {
        this.chatId = chatId;
        this.username = username;
    }

}

File: src/main/kotlin/io/openfuture/openmessanger/repository/entity/ChatParticipant.kt
package io.openfuture.openmessanger.repository.entity

import jakarta.persistence.*

@Entity
@Table(name = "chat_participant")
class ChatParticipant(var chatId: Int?, var username: String?) {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int? = null

}